### ISSUE TITLE
How to get Cali running? (SyntaxError: Unexpected token 'T', "To build a"... is not valid JSON)

### ISSUE THREAD

**Issue by tomsoderlund** [callstackincubator/cali#61]
I start with

    npx cali

I enter my OPENAI_API_KEY, and then answer the question “What do you want to do today?”.

Then I got this:

    ◇  What do you want to do today?
    │  Build a tile drag-and-drop game using react-native-reanimated and
    react-native-gesture-handler
    │
    ◇  Done.
    undefined:1
    To build a tile drag-and-drop game using `react-native-reanimated` and `react-native-gesture-handler`, we need to ensure that these libraries are installed in your project. Let's start by checking if they are already installed. If not, we will proceed to install them.
    ^

    SyntaxError: Unexpected token 'T', "To build a"... is not valid JSON
        at JSON.parse (<anonymous>)
        at file:///Users/tomsoderlund/.npm/_npx/f461be0d0ab30861/node_modules/cali/dist/index.js:554:59
        at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

    Node.js v20.10.0

**Comment by grabbou** [comment]
Yeah, this is very annoying side-effect of Vercel AI SDK not supporting both function calling and structured outputs. There's pending rewrite of internals over to a different AI agents framework that will fix this and a bunch of other issues.

Please stand by for the next two days for a overhaul, hopefully gets the job done!
